.github/workflows/test.yaml

```diff
@@ -149,3 +149,25 @@ jobs:
       container_image: "rapidsai/ci-conda:latest"
       run_script: |
         ci/cudf_pandas_scripts/third-party-integration/test.sh python/cudf/cudf_pandas_tests/third_party_integration_tests/dependencies.yaml
+  wheel-tests-cudf-polars:
+    secrets: inherit
+    uses: rapidsai/shared-workflows/.github/workflows/wheels-test.yaml@branch-24.12
+    with:
+      # This selects "ARCH=amd64 + the latest supported Python + CUDA".
```

Comment:
Good question. It probably makes sense to relax at least the arch and Python filters. The CUDA filter is fine since we have a strong CUDA 12 focus for Polars, at least for now (e.g. that's [the install supported directly from Polars](https://github.com/pola-rs/polars/blob/main/py-polars/pyproject.toml#L76)).

There are two sets of tests, the tests of cudf-polars itself (i.e. our internals) and the polars test suite run with cudf-polars enabled. The former is what we originally had enabled, and that is very cheap. [Looking at a recent run, that job takes only about 2 mins](https://github.com/rapidsai/cudf/actions/runs/11806536947/job/32892817728?pr=17304). Definitely no problem scaling that up. The Polars test suite is more expensive. It'd still be cheaper than a lot of our other jobs, but [from a recent run it looks to be about 15 mins](https://github.com/rapidsai/cudf/actions/runs/11806536947/job/32892817392?pr=17304). How do we feel about scaling that up? Worth the cost? Should we bifurcate CI vs nightly testing? I'd lean towards enabling all the tests and seeing how it affects CI, but I could easily be convinced otherwise.

Reply:
I would just enable all the tests.

We are focused on CUDA 12 but I think we rely on CUDA 11 jobs in our CI matrix to get full coverage of other dimensions (OS, Python, arch). If we see issues with runtime or test failures in unsupported configurations, we can easily narrow it later.

Reply:
Hmm in #15191 did we intend to limit test jobs as well as build jobs? This feels like an indication that we need to resolve https://github.com/rapidsai/build-planning/issues/43, at least on the wheels side to support multiple test runs from a single build. We should handle cudf-polars the same way that we handle dask-cudf, which ideally means building a single wheel but running on the full matrix (up to resource constraints), but our shared workflows will (unless something has changed) still break if you try that due to trying to download nonexistent artifacts. We don't want to build separate pure Python wheels on each of these though.

If you agree that this issue is broader than cudf-polars, I suggest that we merge this PR and look into resolving rapidsai/build-planning#43 (again, we could skip conda work at first and just get wheels working, although we should address the conda part too eventually) in a follow-up since that will impact many different projects.